packages/annotations/src/ui-components/area.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+  <!-- TODO: use fragments in Vue 3 to avoid insertion of an empty <div> into the layout -->
+  <div>
+    <div class="alpheios-annotations-container" :data-annotation-mode="annotationMode"
+         @click="clickHandler" v-if="annotationMode">
+      <slot></slot>
+    </div>
+    <template v-else>
+      <slot></slot>
+    </template>
+  </div>
+</template>
+<script>
+import Vue from '@vue-runtime'
+import AnnotationComment from '@annotations/ui-components/comment.vue'
+import AnnotationDefinition from '@annotations/ui-components/definition.vue'
+import AnnotationInflection from '@annotations/ui-components/inflection.vue'
+
+const COMPONENT_ID = 'annotationComponent'
+const ANNOTATABLE_ATTR_NAME = 'data-annotation-type'
+const SELECTED_ATTR_NAME = 'data-annotation-selected'
+const CONTAINER_ATTR_NAME = 'data-annotation-container'
+const AnnotationTypes = {
+  COMMENT: 'comment',
+  DEFINITION: 'definition',
+  INFLECTION: 'inflection',
+  LEMMA: 'lemma',
+  WORDFORM: 'wordForm'
+}
+const EMPTY_SELECTION = {
+  isSelected: false,
+  target: null,
+  mountContainer: null,
+  vueInstance: null
+}
+
+export default {
+  name: 'AnnotationArea',
+  selection: Object.assign({}, EMPTY_SELECTION),
+  props: {
+    annotationMode: {
+      type: Boolean,
+      required: false,
+      default: false
+    }
+  },
+
+  methods: {
+    clickHandler: function (evt) {
+      const target = evt.target.closest(`[${ANNOTATABLE_ATTR_NAME}]`)
```

Comment:
This attribute serves as an indicator of what elements would be allowed to add/edit/remove annotations. Let's take. for example, https://github.com/alpheios-project/alpheios-core/blob/dev-annotation-ui-next/packages/components/src/vue/components/morph-parts/principal-parts.vue#L14. The presence of the `data-annotation-type` is an indication that this element must have annotations related functionality. The value, `comment` specifies what type of annotation functionality should be available.

My goal is to make the annotations UI as unobtrusive as possible. The source code where annotation functionality to be added should require as little modifications as possible. That's why the use of data attributes. Thy do not require components that use them to import anything. Only one component is necessary: the root `AnnotationArea`  component that should wrap the code with the `data-annotation` tags. See https://github.com/alpheios-project/alpheios-core/blob/dev-annotation-ui-next/packages/components/src/vue/components/morph.vue#L3 as an example. 

This component, once activated, will listen for the clicks within its inner content. Then, if a click comes from an element whose parent has a `data-annotation-type` attribute, it assumes the element is in the edit mode (or switches from the edit mode to the normal one), and it creates a Vue component that represents an editing panel, and inserts it below the element. Annotation package is also responsible for bringing in the CSS for the annotatable elements.

The only part I'm not quite happy with are panels to add new item types (lemma, inflections, etc.). Right now they require corresponding annotation components to be imported, as in https://github.com/alpheios-project/alpheios-core/blob/dev-annotation-ui-next/packages/components/src/vue/components/morph.vue#L23.

I think ideally there should be the code like `data-annotationtype="lemma"` and, when annotation root component is mounted, it should be replaced with the component's code. The problem here, however, is that the `AnnotationArea` component is mounted before the `morph` one and thus it is too early for the `AnnotationArea` to access to HTML code of the `morph`'s template. Not sure how deal with it. Maybe to have one generic component for adding panels like this that will insert the corresponding panel inside itself. This would eliminate the need of the client to import different kinds of components. It is enough for it to know and import only one component type.

Please let me know if this makes sense.

Reply:
I like the approach. I think it's an interesting question where the knowledge about which component to import should go.  On the one hand, it's appealing to have that all encapsulated in the annotation package, but on the other hand, the domain knowledge about how a specific data object should be annotated probably belongs with the domain component.   I think this will become clearer as we design the graphQL api for the mutations.